Web Extension Transformer: esmodule-helpers.js fails to load

- Follows https://github.com/parcel-bundler/parcel/pull/7050

# 🐛 bug report

My options page isn't being built correctly. Opening the options page in the browser results into this line failing:

```js
var parcelHelpers = require("@parcel/transformer-js/src/esmodule-helpers.js");
```

and the console error:

```
options.6f3d3386.js:63 Uncaught Error: Cannot find module '8ISrk'
    at newRequire (options.6f3d3386.js:61:19)
    at newRequire (options.6f3d3386.js:45:18)
    at localRequire (options.6f3d3386.js:84:35)
    at Object.3fApi.webextension-polyfill (options.6f3d3386.js:146:21)
    at newRequire (options.6f3d3386.js:71:24)
    at options.6f3d3386.js:122:5
    at options.6f3d3386.js:145:3
```

## 🎛 Configuration (.babelrc, package.json, cli command)

<!--- If describing a bug, tell us what your babel configuration looks like -->

```json
{
	"extends": "@parcel/config-webextension"
}
```
```json
	"browserslist": [
		"last 1 Chrome version",
		"last 1 Firefox version"
	],
```

## 💻 Code Sample


The content of `options.js` is just:


```js
import browser from 'webextension-polyfill';
```

and part of the output is:

```js
})({"3fApi":[function(require,module,exports) {
var parcelHelpers = require("@parcel/transformer-js/src/esmodule-helpers.js");
var _webextensionPolyfill = require("webextension-polyfill");
var _webextensionPolyfillDefault = parcelHelpers.interopDefault(_webextensionPolyfill);

},{"webextension-polyfill":"kPFTQ","@parcel/transformer-js/src/esmodule-helpers.js":"8ISrk"}],"kPFTQ":[function(require,module,exports) {
```

Full example in https://github.com/fregante/browser-extension-template/pull/75/files

## 🌍 Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.1049
| parcel/config-webextension | 2.4.2-nightly.2674 |
| Node             | v16.4.2
| npm/Yarn         | 7.18.1
| Operating System | macOS 11

--- a/packages/packagers/webextension/src/WebExtensionPackager.js
+++ b/packages/packagers/webextension/src/WebExtensionPackager.js
@@ -42,22 +42,26 @@ export default (new Packager({
         ),
       ];
 
-      war.push({
-        matches: contentScript.matches,
-        extension_ids: [],
-        resources: jsBundles
-          .flatMap(b => {
-            const children = [];
-            const siblings = bundleGraph.getReferencedBundles(b);
-            bundleGraph.traverseBundles(child => {
-              if (b !== child && !siblings.includes(child)) {
-                children.push(child);
-              }
-            }, b);
-            return children;
-          })
-          .map(relPath),
-      });
+      const resources = jsBundles
+        .flatMap(b => {
+          const children = [];
+          const siblings = bundleGraph.getReferencedBundles(b);
+          bundleGraph.traverseBundles(child => {
+            if (b !== child && !siblings.includes(child)) {
+              children.push(child);
+            }
+          }, b);
+          return children;
+        })
+        .map(relPath);
+
+      if (resources.length > 0) {
+        // In the future, maybe use "matches" as well
+        war.push({
+          extension_ids: [],
+          resources,
+        });
+      }
     }
     manifest.web_accessible_resources = (
       manifest.web_accessible_resources || []

--- a/packages/transformers/webextension/src/WebExtensionTransformer.js
+++ b/packages/transformers/webextension/src/WebExtensionTransformer.js
@@ -99,6 +99,7 @@ async function collectDependencies(
           assets[j] = asset.addURLDependency(assets[j], {
             // This causes the packager to re-run when these assets update
             priority: 'parallel',
+            bundleBehavior: 'isolated',
             loc: {
               filePath,
               ...getJSONSourceLocation(
@@ -190,6 +191,8 @@ async function collectDependencies(
           await glob(path.join(assetDir, files[j]), fs, {})
         ).map(fp =>
           asset.addURLDependency(path.relative(assetDir, fp), {
+            priority: 'parallel',
+            bundleBehavior: 'isolated',
             needsStableName: true,
             loc: {
               filePath,
@@ -224,6 +227,8 @@ async function collectDependencies(
     const obj = parent[lastLoc];
     if (typeof obj == 'string')
       parent[lastLoc] = asset.addURLDependency(obj, {
+        priority: 'parallel',
+        bundleBehavior: 'isolated',
         loc: {
           filePath,
           ...getJSONSourceLocation(ptrs[location], 'value'),
@@ -233,6 +238,8 @@ async function collectDependencies(
     else {
       for (const k of Object.keys(obj)) {
         obj[k] = asset.addURLDependency(obj[k], {
+          priority: 'parallel',
+          bundleBehavior: 'isolated',
           loc: {
             filePath,
             ...getJSONSourceLocation(ptrs[location + '/' + k], 'value'),
@@ -247,6 +254,8 @@ async function collectDependencies(
       program.background.page = asset.addURLDependency(
         program.background.page,
         {
+          priority: 'parallel',
+          bundleBehavior: 'isolated',
           loc: {
             filePath,
             ...getJSONSourceLocation(ptrs['/background/page'], 'value'),
@@ -285,6 +294,8 @@ async function collectDependencies(
       program.background.service_worker = asset.addURLDependency(
         program.background.service_worker,
         {
+          priority: 'parallel',
+          bundleBehavior: 'isolated',
           loc: {
             filePath,
             ...getJSONSourceLocation(

--- a/packages/transformers/webextension/src/schema.js
+++ b/packages/transformers/webextension/src/schema.js
@@ -91,7 +91,7 @@ const commonProps = {
     additionalProperties: {
       type: 'object',
       properties: {},
-    }
+    },
   },
   chrome_settings_overrides: {
     type: 'object',

--- a/packages/transformers/webextension/src/schema.js
+++ b/packages/transformers/webextension/src/schema.js
@@ -85,6 +85,14 @@ const commonProps = {
   description: string,
   icons,
   author: string,
+  browser_specific_settings: {
+    type: 'object',
+    properties: {},
+    additionalProperties: {
+      type: 'object',
+      properties: {},
+    }
+  },
   chrome_settings_overrides: {
     type: 'object',
     properties: {

